spike_main/spike.cc

```diff
@@ -188,7 +190,20 @@ static std::vector<mem_cfg_t> parse_mem_layout(const char* arg)
       exit(EXIT_FAILURE);
     }
 
-    res.push_back(mem_cfg_t(reg_t(base), reg_t(size)));
+    const unsigned long long max_allowed_pa = (1ull << MAX_PADDR_BITS) - 1ull;
+    assert(max_allowed_pa <= std::numeric_limits<reg_t>::max());
+    mem_cfg_t mem_region(base, size);
+    if ((mem_region.base + mem_region.size - 1) > max_allowed_pa) {
+      fprintf(stderr, "Unsupported memory region "
+                      "{base = 0x%" PRIx64 ", size = 0x%" PRIx64 "} specified\n"
+                      "    The largest accessible physical address "
+                      "is 0x%llX (defined by MAX_PADDR_BITS constant)\n",
```

Comment:
> "Redundant" is misspelled
My bad. fixed.

> Be consistent in usage of lower vs uppercase hex digits.
Done.

> I also wonder if we could make the error message a little easier to read

Sure, however I'd like to use `__builtin_clzll` to calculate the number of required physical bits. Do you think it's appropriate? We already use `__builtin_expect` in likely/unlikely macro, so probably this should be ok.. Still I want to double-check

Reply:
I'm okay with that, since it works in both GCC and clang. @aswaterman ?

Reply:
I've been trying to avoid that kind of thing.  Note that we only use `__builtin_expect` if `__GNUC__` is defined.

Please use this one: https://github.com/riscv-software-src/riscv-isa-sim/blob/ab3225a3ff687fda8b4180f9e4e0949a400d1247/riscv/arith.h#L157

Reply:
@aswaterman , 
> Please use this one:
Thanks!

> I also wonder if we could make the error message a little easier to read, instead of the user having to count all those digits? 

Addressed